--- contrib/Makefile.in.orig	2010-07-09 19:34:33.000000000 -0700
+++ contrib/Makefile.in	2010-07-09 19:34:40.000000000 -0700
@@ -7,7 +7,7 @@
 # by default make sublibs. the functionparser is not part of the
 # default make list, since it needs to be built *before* the rest of
 # this stuff as it is needed by libbase
-default: hsl umfpack functionparser tbb
+default: hsl functionparser
 
 
 ifeq ($(USE_CONTRIB_HSL),yes)
