--- deal.II/source/numerics/mesh_worker_info.cc	(revision 21391)
+++ deal.II/source/numerics/mesh_worker_info.cc	(revision 21392)
@@ -29,7 +29,6 @@
   template class IntegrationInfo<deal_II_dimension, deal_II_dimension>;
   template class IntegrationInfoBox<deal_II_dimension, deal_II_dimension>;
   
-  template class LocalResults<float>;
   template class DoFInfo<deal_II_dimension,deal_II_dimension,float>;
   template class DoFInfoBox<deal_II_dimension,
 			    DoFInfo<deal_II_dimension,deal_II_dimension,float> >;
@@ -37,7 +36,6 @@
   template void IntegrationInfo<deal_II_dimension>::fill_local_data(
     const DoFInfo<deal_II_dimension, deal_II_dimension, float>&, bool);
   
-  template class LocalResults<double>;
   template class DoFInfo<deal_II_dimension,deal_II_dimension,double>;
   template class DoFInfoBox<deal_II_dimension,
 			    DoFInfo<deal_II_dimension,deal_II_dimension,double> >;
